### PR TITLE
NekRS

### PR DESCRIPTION
**Pull request by liruipeng**
This PR adds the optimizations in IJ solvers that were done for NekRS and could be good for others as well. 

1. It introduces `HYPRE_IJVectorUpdateValues` to update IJ vectors which is less expensive (and more restrictive in the usage) than `SetValues` and `AddToValues`. 
2. It adds an option in `hypre_BoomerAMGRelax` to indicate whether the input initial vector (`u`) is zero, so that a MATVEC can be saved per relaxation. 
3. It adds `HYPRE_WITH_GPU_AWARE_MPI` option to CMake.
4. It adds more tests to the regression and updates the results

- [x] tux
- [x] lassen
- [x] tioga
- [ ] jlse

### CODE REVIEW

File: src/config/HYPRE_config.h.cmake.in [hunk 1]
```diff
@@ -35,6 +35,8 @@
 /* Use cudaMallocAsync */
 #cmakedefine HYPRE_USING_DEVICE_MALLOC_ASYNC 1
 
+#cmakedefine HYPRE_WITH_GPU_AWARE_MPI 1
```
Comment (victorapm) [inline src/config/HYPRE_config.h.cmake.in:39]: Very minor change:

```suggestion
/* Use GPU-aware MPI */
#cmakedefine HYPRE_WITH_GPU_AWARE_MPI 1
```

**Review by victorapm — APPROVED** [review]
Thanks! These are very nice changes. Will it be necessary to update the IJ benchmark timings? I assume BoomerAMG is faster now.

**Review by osborn9 — APPROVED** [review]

**Comment by osborn9** [comment]
Looks good (esp. from the CMake perspective). Thanks!

**Review by ulrikeyang — APPROVED** [review]
This looks good. Thanks

File: src/parcsr_mv/_hypre_parcsr_mv.h [hunk 2]
```diff
@@ -246,6 +246,8 @@ typedef struct hypre_ParVector_struct
 
    /* Does the Vector create/destroy `data'? */
    HYPRE_Int             owns_data;
+   /* If the vector is all zeros */
+   HYPRE_Int             all_zero;
```
Comment (victorapm) [inline src/parcsr_mv/_hypre_parcsr_mv.h]: Very minor: should we call it `all_zeros`? And consequently `hypre_ParVectorAllZeros`. This change looks very nice!

File: src/parcsr_ls/par_relax.c [hunk 3]
```diff
@@ -1119,7 +1118,7 @@ hypre_BoomerAMGRelax7Jacobi( hypre_ParCSRMatrix *A,
    /*-----------------------------------------------------------------
     * Perform Matvec Vtemp = w * (f - Au)
     *-----------------------------------------------------------------*/
-   if (zero_u)
+   if (hypre_ParVectorAllZero(u))
```
Comment (victorapm) [inline src/parcsr_ls/par_relax.c]: How about moving this check inside `hypre_ParCSRMatrixMatvec`? Then, other parts of the code could potentially benefit from the fact that a vector is all zeros.

**Review by victorapm — APPROVED** [review]
This looks nice! The `all_zeros` variable makes the code cleaner and easier to follow. Thanks!! I added one suggestion that might be interesting to consider implementing and a minor comment

**Review by ulrikeyang — COMMENTED** [review]
This looks fine. Thanks

File: src/parcsr_ls/par_relax.c (hunk 3 above)
Comment (liruipeng) [reply src/parcsr_ls/par_relax.c]: let's talk about this in the meeting later. Thanks!

**Pull request merged**